e2e/support/cypress.js

```diff
@@ -114,14 +114,29 @@ Cypress.on("window:load", window => {
   };
 });
 
-// cypress-terminal-report
 if (isCI) {
+  // cypress-terminal-report
   afterEach(() => {
     cy.wait(50, { log: false }).then(() =>
       cy.addTestContext(Cypress.TerminalReport.getLogs("txt")),
     );
   });
 
+  // Fast failure notifications
+  afterEach(() => {
+    const testState = Cypress.mocha.getRunner().suite.ctx.currentTest.state;
+    const testInfo = Cypress.mocha.getRunner().suite.ctx.currentTest;
+
+    const isLastRetry = testInfo.currentRetry() === testInfo.retries();
+
+    if (testState === "failed" && isLastRetry) {
+      cy.task("reportCIFailure", {
+        spec: Cypress.spec,
+        test: Cypress.currentTest,
```

Comment:
Out of curiosity - is `Cypress.currentTest` different from `Cypress.mocha.getRunner().suite.ctx.currentTest`?

Reply:
yes 🥴